### PR TITLE
Global installable hook

### PR DESCRIPTION
**Pull request by 0atman**
Global hook ready for testing

todo: install script.

### CODE REVIEW

File: global_install/hooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python
```
Comment (0atman) [inline global_install/hooks/pre-commit]: This file is mostly  a copy of the `precommit framework` file. Don't @-me about the bad python. yet.

File: README.md [hunk 2]
```diff
@@ -2,10 +2,9 @@
 
 ```shell
 pip install pre-commit detect-secrets
-mkdir .gds
-detect-secrets scan > .gds/.secrets.baseline
-detect-secrets audit .gds/.secrets.baseline
-pre-commit install -c .gds/.pre-commit-config.yaml #this file is from this repo
+mkdir ~/gds
```
Comment (danjoneslf) [inline README.md]: What happens if the user already has a folder called ~/gds? I would have thought that's quite likely. Maybe we should call it ~/gds-pre-commit ~/gds/github-pre-commit or something that's not likely to be used for something else. If you're creating folder structure within that then maybe it's just a mkdir -p so it doesn't fail if it's already there.

File: global_install/hooks/pre-commit [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+if sys.version_info < (3, 7):  # https://bugs.python.org/issue25942
+    def _subprocess_call(cmd):  # this is the python 2.7 implementation
+        return subprocess.Popen(cmd).wait()
+else:
+    _subprocess_call = subprocess.call
+
+
+def main():
+    config_path = os.path.join(top_level, ".git/config")
```
Comment (danjoneslf) [inline global_install/hooks/pre-commit]: So this is run per repo looking for alphagov in the .git/config. What runs this?

**Review by danjoneslf — COMMENTED** [review]
Left a couple of comments.

File: global_install/hooks/pre-commit (hunk 3 above)
Comment (0atman) [reply global_install/hooks/pre-commit]: it's a global precommit hook!

File: README.md (hunk 2 above)
Comment (0atman) [reply README.md]: yeah, we're gonna parameterize this

File: README.md [hunk 4]
```diff
@@ -2,10 +2,9 @@
...
+git clone https://github.com/alphagov/gds-pre-commit.git ~/gds/gds-pre-commit
+git config --global core.hooksPath $HOME/gds/gds-pre-commit/global_install/hooks
```
Comment (danjoneslf) [inline README.md]: Also what if $HOME/gds/gds-pre-commit is where the user has checked out that code to already?

File: global_install/hooks/pre-commit (hunk 1 above)
Comment (0atman) [reply global_install/hooks/pre-commit]: Having read more of the python, it's pretty good, for cross-platform resilience.

File: README.md (hunk 2 above)
Comment (0atman) [reply README.md]: Actually, we've moved do `.gds-pre-commit/`

**Comment by akinnane** [comment]
I've tested this and it WGFM!

**Review by akinnane — APPROVED** [review]

**Pull request merged**